blog/2022-12-07-build-a-frontend-dapp-for-celo-network-in-angular/2022-12-07-build-a-frontend-dapp-for-celo-network-in-angular.md

```diff
@@ -0,0 +1,1698 @@
+
+    </div>
+  </div>
+</nav>
+<div class="container-fluid py-3">
+  <div class="row my-3">
+    <div class="col d-flex justify-content-center">
+      <h3>My Angular DAPP</h3>
+
+    </div>
+  </div>
+  <div class="row my-3">
+    <div class="col d-flex justify-content-center">
+      <div *ngIf="accounts "> <span>{{balance}} CELO </span> {{accounts[0]}}</div>
+      <button *ngIf="!accounts " class="btn btn-primary py-2 px-4 rounded" (click)="Connect()" >
+          Connect your wallet
+      </button>
+    </div>
+  </div>
+
+  <div class="row">
+    <div class="col my-3 ">
+      <router-outlet></router-outlet>
+    </div>
+  </div>
+</div>
+```
+
+
+
+
+Now open the app.module file(**src/app/app.module.ts**), let's configure our app module to support routing. Add this import to the file (under existing imports)
+
+```js
+import {RouterModule, Routes } from '@angular/router';
+```
+
+Also create an array for storing our routes
+
+```js 
+const routes: Routes = [];
+```
+
+Finally, in your NgModule, add the router module to the imports
+
+```js
+@NgModule({
+	...
+  imports: [
+	....
+	RouterModule.forRoot(routes)
+	],
+  ...
+})
+export class AppModule {}
+```
+
+
+Your App Module should look like this now
+
+```js
+import { NgModule } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+
+import { AppComponent } from './app.component';
+import { RouterModule, Routes } from '@angular/router'
+
+const routes: Routes = [];
+
+@NgModule({
+  declarations: [
+    AppComponent
+  ],
+  imports: [
+    BrowserModule,
+    BrowserAnimationsModule,
+	RouterModule.forRoot(routes)
+  ],
+  providers: [],
+  bootstrap: [AppComponent]
+})
+export class AppModule { }
+```
+
+
+Let's add a new Home Component and setup a route for it. We can do this from the Angular-cli, so run this on your terminal (from the root angular folder path)
+
+```text
+ng g c home
+```
+
+
+Open the newly added Home view (/src/app/home/home.html) and update it to
+
+```html
+<div class="row">
+    <div class="col">
+        <h3>Welcome to our Auction Site</h3>
+    </div>
+</div>
+```
+
+Then add a route for your home page (update the routes array in your app.module)
+
+```js
+const routes: Routes = [
+  { path: 'home', component: HomeComponent },
+  { path: '', redirectTo: '/home', pathMatch: 'full' },
+]
+```
+
+
+
+Now, check your running application in your browser, it should still be working, good. And now it has a nav menu, and some page structure, great.
+
+
+## Display Connected Wallet Information
+
+
+So far, our connect button is working and shows us the account it's connected to, we will extend this a bit.
+
+Go back to the app.component.ts file
+
+Change the ***Connect*** method to an async method, rename the ***data*** class variable to ***accounts*** and update the ***Connect*** method to reflect this.
+
+` `**~~`data`~~**` accounts: any[]|undefined;`
```

Comment:
What's happening here?

Reply:
I showed the renaming of the variable, i think i will just remove that